crates/router/src/connector/bankofamerica/transformers.rs

```diff
@@ -221,6 +280,301 @@ pub struct BillTo {
     email: pii::Email,
 }
 
+impl TryFrom<&types::SetupMandateRouterData> for BankOfAmericaPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::SetupMandateRouterData) -> Result<Self, Self::Error> {
+        let email = item.request.get_email()?;
+        let bill_to = build_bill_to(item.get_billing()?, email)?;
+
+        let order_information = OrderInformationWithBill {
+            amount_details: Amount {
+                total_amount: "0".to_string(),
+                currency: item.request.currency,
+            },
+            bill_to,
+        };
+        let (action_list, action_token_types, authorization_options) = (
+            Some(vec![BankOfAmericaActionsList::TokenCreate]),
+            Some(vec![BankOfAmericaActionsTokenType::PaymentInstrument]),
+            Some(BankOfAmericaAuthorizationOptions {
+                initiator: Some(BankOfAmericaPaymentInitiator {
+                    initiator_type: Some(BankOfAmericaPaymentInitiatorTypes::Customer),
+                    credential_stored_on_file: Some(true),
+                    stored_credential_used: None,
+                }),
+                merchant_intitiated_transaction: None,
+            }),
+        );
+
+        let client_reference_information = ClientReferenceInformation {
+            code: Some(item.connector_request_reference_id.clone()),
+        };
+
+        let merchant_defined_information = item.request.metadata.clone().map(|metadata| {
+            Vec::<MerchantDefinedInformation>::foreign_from(metadata.peek().to_owned())
+        });
+
+        let (
+            payment_information,
+            solution,
+            commerce_indicator,
+            consumer_authentication_information,
+        ) = match item.request.payment_method_data.clone() {
+            api::PaymentMethodData::Card(ccard) => {
```

Comment:
Break the impl for cards and wallets into smaller from & try_from. Keep the code modular and readable